### PR TITLE
Add an initial current working directory API.

### PR DESCRIPTION
**Pull request by sunfishcode**
Add a function for returning an initial current working directory.

### CODE REVIEW

**Comment by codefromthecrypt** [comment]
appreciate the work on this. It would be nice to have this rendered in the canonical ABI or old witx so I can see the signature vs trying to calculate it in my head. Things are more abstract which has benefits, but it would also be nice to know what concretely this would become without mental transpiling.

**Review by pchickey — COMMENTED** [review]
for components provided an empty set of preopens (i.e., the filesystem is required at link time, but at run time the world has decided not to provide any usable filesystem descriptors), this interface is not satisfiable - so instead, I think it needs to return an `option<tuple<descriptor, string>>`.

**Comment by sunfishcode** [comment]
> for components provided an empty set of preopens (i.e., the filesystem is required at link time, but at run time the world has decided not to provide any usable filesystem descriptors), this interface is not satisfiable - so instead, I think it needs to return an `option<tuple<descriptor, string>>`.

Good point; fixed.

File: command.md [hunk 1]
```diff
@@ -2943,6 +2943,13 @@ values each time it is called.</p>
 <ul>
 <li><a name="get_directories.0"></a> list&lt;(<a href="#descriptor"><a href="#descriptor"><code>descriptor</code></a></a>, <code>string</code>)&gt;</li>
 </ul>
+<h4><a name="initial_cwd"><code>initial-cwd: func</code></a></h4>
+<p>Return a preopen for use as the program's initial current working
```
Comment (codefromthecrypt) [inline command.md:2947]: I wonder why we are returning a pre-open and what I believe to be the path, instead of just the path. It seems the guest will need to resolve pre-opens for path resolution anyway. Adding a pre-open FD here seems to give a chance or at least a question of why resolution of the pwd would happen in a different way that a normal path lookup in the guest. It also forces the host to search the host-side pre-opens for that dir which feels unnecessary.

TL;DR' what's motivating the descriptor part of the tuple and what would happen if it wasn't here?

Comment (sunfishcode) [reply command.md:2947]: No strong reason; the cwd in POSIX acts a lot like a file descriptor that the OS holds for you, so it seemed natural to say there should be a file descriptor here too. But you're right, if someone wants that, they can pass in a regular preopen for it. I've now removed the descriptor here.
